lambda/wrapper.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Specify the noexecwrapper build tag to remove the wrapper tampoline from
+// this library if it is undesirable.
+//go:build unix && !noexecwrapper
+// +build unix,!noexecwrapper
+
+package lambda
+
+import (
+	"log"
+	"os"
+	"syscall"
+)
+
+const awsLambdaExecWrapper = "AWS_LAMBDA_EXEC_WRAPPER"
+
+func init() {
```

Comment:
I think it'd be problematic to introduce this as an `init()` function. As functions with possibly vestigial definitions of AWS_LAMBDA_EXEC_WRAPPER would see a behavior change.

So if this was to be added, I think it'd best to do it as a `lambda.Option`. eg: `lambda.StartWithOptions(..., lambda.WithEnableExecWrapper())`

Reply:
> So if this was to be added, I think it'd best to do it as a lambda.Option. eg: lambda.StartWithOptions(..., lambda.WithEnableExecWrapper())

I like this idea a lot!